src/types/src/dds_tt.h

```diff
@@ -0,0 +1,214 @@
+ * 0-7: the specific base type
+ * 8-13: the type spec
+ * 14-15: the definitions
+ * 16-: the other nodes
+ */
+
+#define DDS_TS_TYPE_SPEC(X)             (((X)&7)<<8)
+#define DDS_TS_IS_TYPE_SPEC(T)          (((7<<8)&(T)) != 0)
+#define DDS_TS_BASE_TYPE(X)             (DDS_TS_TYPE_SPEC(1) | (X))
+#define DDS_TS_IS_BASE_TYPE(T)          (((7<<8)&(T)) == 1)
+#define DDS_TS_GET_BASE_TYPE(T)         (((7<<8)|31)&(T))
+#define DDS_TS_SHORT_TYPE               DDS_TS_BASE_TYPE(1)
+#define DDS_TS_LONG_TYPE                DDS_TS_BASE_TYPE(2)
+#define DDS_TS_LONG_LONG_TYPE           DDS_TS_BASE_TYPE(3)
+#define DDS_TS_UNSIGNED_SHORT_TYPE      DDS_TS_BASE_TYPE(4)
+#define DDS_TS_UNSIGNED_LONG_TYPE       DDS_TS_BASE_TYPE(5)
+#define DDS_TS_UNSIGNED_LONG_LONG_TYPE  DDS_TS_BASE_TYPE(6)
+#define DDS_TS_CHAR_TYPE                DDS_TS_BASE_TYPE(7)
+#define DDS_TS_WIDE_CHAR_TYPE           DDS_TS_BASE_TYPE(8)
+#define DDS_TS_BOOLEAN_TYPE             DDS_TS_BASE_TYPE(9)
+#define DDS_TS_OCTET_TYPE               DDS_TS_BASE_TYPE(10)
+#define DDS_TS_INT8_TYPE                DDS_TS_BASE_TYPE(11)
+#define DDS_TS_UINT8_TYPE               DDS_TS_BASE_TYPE(12)
+#define DDS_TS_FLOAT_TYPE               DDS_TS_BASE_TYPE(13)
+#define DDS_TS_DOUBLE_TYPE              DDS_TS_BASE_TYPE(14)
+#define DDS_TS_LONG_DOUBLE_TYPE         DDS_TS_BASE_TYPE(15)
+#define DDS_TS_FIXED_PT_CONST_TYPE      DDS_TS_BASE_TYPE(16)
+#define DDS_TS_ANY_TYPE                 DDS_TS_BASE_TYPE(17)
+#define DDS_TS_SEQUENCE_TYPE            DDS_TS_TYPE_SPEC(2)
+#define DDS_TS_STRING_TYPE              DDS_TS_TYPE_SPEC(3)
+#define DDS_TS_WIDE_STRING_TYPE         DDS_TS_TYPE_SPEC(4)
+#define DDS_TS_FIXED_PT_TYPE            DDS_TS_TYPE_SPEC(5)
+#define DDS_TS_MAP_TYPE                 DDS_TS_TYPE_SPEC(6)
+#define DDS_TS_DEFINITION(X)            (((X)&15)<<12)
+#define DDS_TS_IS_DEFINITION(T)         (((15<<12)&(T)) != 0)
+#define DDS_TS_MODULE                   DDS_TS_DEFINITION(1)
+#define DDS_TS_FORWARD_STRUCT           DDS_TS_DEFINITION(2)
+#define DDS_TS_STRUCT                   DDS_TS_DEFINITION(3)
+#define DDS_TS_DECLARATOR               DDS_TS_DEFINITION(4)
+#define DDS_TS_OTHER(X)                 ((X)<<16)
+#define DDS_TS_STRUCT_MEMBER            DDS_TS_OTHER(1)
+#define DDS_TS_ARRAY_SIZE               DDS_TS_OTHER(2)
+
+/* Open issues:
+ * - include file and line information in type definitions.
+ */
+
+typedef char *dds_ts_identifier_t;
+
+/* Literals
+ *
+ * Literals are values, either stated or calculated with an expression, that
+ * appear in the IDL declaration. The literals only appear as members of
+ * IDL elements, such as the constant definition and the case labels.
+ */
+
+typedef uint32_t dds_ts_node_flags_t;
+
+typedef struct {
+  dds_ts_node_flags_t flags; /* flags defining the kind of the literal */
+  union {
+    bool bln;
+    char chr;
+    unsigned long wchr;
+    char *str;
+    unsigned long long ullng;
+    signed long long llng;
+    long double ldbl;
+  };
+} dds_ts_literal_t;
```

Comment:
This type was already defined somewhere else. I get the feeling that the same goes for multiple types, but haven't checked. Is there a special reason this is defined twice? It seems wrong...

Reply:
I have removed the type.h file, an older version of dds_tt.h, that got included in the commit.